ufl/form.py

```diff
@@ -69,7 +69,149 @@ def _sorted_integrals(integrals):
     return tuple(all_integrals)  # integrals_dict
 
 
-class Form(object):
+class BaseForm(object):
```

Comment:
So the idea here is that where previously a form was always a list of integrals, now a form can be a linear combination of lists of integrals, linear operators, and coarguments?

What is the consequence (if any) for `compute_form_data` which does an analysis of `Form` objects to determine what needs to be assembled by the problem solving environment?

Reply:
`Form`s are unchanged, so no consequences for `compute_form_data`. The problem solving environment needs to handle the rest (or not).